### PR TITLE
[ACCOUNT-2501] refactor: getValidToken replacement for getOrRefreshToken

### PR DESCRIPTION
**Pull request by hschoenenberger**
…oken

### CODE REVIEW

File: src/Hook/ActionObjectShopUpdateAfter.php [hunk 1]
```diff
@@ -31,50 +31,54 @@ class ActionObjectShopUpdateAfter extends Hook
...
      *
      * @return bool
-     *
-     * @throws Exception
      */
     public function execute(array $params = [])
     {
-        /** @var Link $link */
-        $link = $this->module->getService(Link::class);
+        $this->updateUserShop($params['object']);
```
Comment (atourneriePresta) [inline src/Hook/ActionObjectShopUpdateAfter.php:37]: Maybe reinstanciate a Shop object as before to be consistent and not depending on prestashop for it (difference between version, inconsistency, ...)

File: tests/TestCase.php [hunk 2]
```diff
@@ -209,20 +216,56 @@ protected function createMockWithMethods($class, $methods = [])
...
+     * @param mixed $object
+     * @param string $propertyName
+     * @param mixed $replacement
      *
      * @return void
      *
      * @throws \ReflectionException
      */
-    protected function replaceDependency($classInstance, $dependencyName, $newDependency)
+    protected function replaceProperty($object, $propertyName, $replacement)
```
Comment (atourneriePresta) [inline tests/TestCase.php:229]: Disgusting

**Review by atourneriePresta — CHANGES_REQUESTED** [review]

File: src/Hook/ActionObjectShopUpdateAfter.php (hunk 1 above)
Comment (hschoenenberger) [reply src/Hook/ActionObjectShopUpdateAfter.php:37]: I think that's not necessary as long as the associated feature test is passing.

File: src/Account/Session/Session.php [hunk 3]
```diff
@@ -33,11 +33,26 @@ abstract class Session implements SessionInterface
     protected $refreshTokenErrors = [];
 
     /**
+     * @deprecated
```
Comment (jokesterfr) [inline src/Account/Session/Session.php:36]: in this deprecation notice, you shall mention what public method the user must use instead

File: src/Account/Session/SessionInterface.php [hunk 4]
```diff
@@ -50,6 +50,8 @@ public function setToken($token, $refreshToken = null);
     public function refreshToken($refreshToken = null);
 
     /**
+     * @deprecated
```
Comment (jokesterfr) [inline src/Account/Session/SessionInterface.php:53]: same as above, give a hint to the user so he/she would know what to use instead

**Review by jokesterfr — APPROVED** [review]

**Pull request merged**